### PR TITLE
Update configuration for demo dashboard

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Tidy up some of the data generation code snippets
- Improve demo dashboard
  - Move homepage to its own icon
  - Format numbers in data table for benchmark analysis page
  - Improve layout of benchmark analysis page (e.g. Add parameter, change layout, change title so it's clearer what the user can interact with)

Note: This is just a quick fix to improve the last screen of the demo dashboard. Hence, I didn't keep the yaml, dict, json examples in sync as we still need to align whether we want to keep these for this demo dashboard or not. This will be tackled in another PR after alignment on what we should do with the demo / feature compendium dashboard.

## Screenshot
Before:
Screenshot 2023-12-13 at 14 57 59

After:
Screenshot 2023-12-13 at 14 48 55

## Checklist

- [x] I have not referenced individuals, products or companies in any commits, directly or indirectly
- [x] I have not added data or restricted code in any commits, directly or indirectly
- [x] I have updated the docstring of any public function/class/model changed
- [ ] I have added tests to cover my changes (if applicable)

## Types of changes

- [x] Docs/refactoring (non-breaking change which improves codebase)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by petar-qb — APPROVED** [review]
I like these changes.

Small suggestion: Maybe we should put the Parameter after Filters inside the benchmark page.

**Review by antonymilne — APPROVED** [review]
Looks good!

Do you think we should change the `NavLink` to have some nice icons here rather than the default `filter_1/2/3`? It would make our demo look a bit slicker maybe, but also adds complexity to the example app. Up to you!

File: vizro-core/examples/default/app.py
Comment (maxschulz-COL) [inline vizro-core/examples/default/app.py:20]: I have only just read this code properly for the first time. I think in its current form it is actually largely confusing: We are associating the mean and sum across continents to individual countries again without changing the column names to reflect this, or without eliminating the countries so that we only look a continent and year.

I would probably change this to something like:

```py
df_transformed = df.groupby(by=["continent", "year"]).agg({"lifeExp": "mean", "pop": "sum", "gdpPercap": "mean"}).reset_index()
```

Comment (maxschulz-COL) [reply vizro-core/examples/default/app.py:20]: I think this may be related to the reason why the average shows up weirdly (including being labeled with Zimbabwe here): 
image

Can we check this quickly?

**Review by maxschulz-COL — CHANGES_REQUESTED** [review]
Generally really like it, but I found some off behviour in the transformed df display. Can you check quickly?

File: vizro-core/examples/default/app.py
Comment (huong-li-nguyen) [reply vizro-core/examples/default/app.py:20]: I tried your suggestion above before actually, but it's not the same 😄 If you do above, it will only give you the country results - not the country next to the continent average result.

Comment (huong-li-nguyen) [reply vizro-core/examples/default/app.py:20]: The results also seem coherent for me? 
Screenshot 2023-12-13 at 16 28 35

The chart on the right only makes sense if you filter on one country, so select a cell in the table

Comment (maxschulz-COL) [reply vizro-core/examples/default/app.py:20]: Oh haha ok :)

Hm ok that makes more sense. I guess it's not quite ideal, but I can see why now!

Comment (huong-li-nguyen) [reply vizro-core/examples/default/app.py:20]: Is it good to go for you then? 😄 Would merge after your approval

**Review by maxschulz-COL — APPROVED** [review]

**Pull request merged**
